spqlios/commons.c

```diff
@@ -115,3 +111,35 @@ double internal_accurate_sin(double x) {
   internal_accurate_sincos(&rcos, &rsin, x);
   return rsin;
 }
+
+EXPORT void spqlios_debug_free(void* addr) { free(addr - 64); }
+
+EXPORT void* spqlios_debug_alloc(uint64_t size) { return malloc(size + 64) + 64; }
+
+EXPORT void spqlios_free(void* addr) {
+#ifdef _WIN32
+  _aligned_free(addr);
+#else
+  free(addr);
+#endif
+}
+
+EXPORT void* spqlios_alloc(uint64_t size) {
+#ifdef _WIN32
```

Comment:
I was thinking more something like: 
```c
#ifndef NDEBUG
// in release mode, the function will return 64-bytes aligned memory
your current function
#else
// in debug mode, the function will not necessarily have any particular alignment
// it will also ensure that memory can only be deallocated with spqlios_free()
return spqlios_debug_alloc(size);
#endif
```